source/Audio.cpp

```diff
@@ -0,0 +1,103 @@
+#include <algorithm>
+#include <iostream>
+#include <thread>
+#include <chrono>
+#include "AudioListener.hpp"
+#include "AudioSource.hpp"
+#include "Audio.hpp"
+#include "Music.hpp"
+#include "Vect.hpp"
+#include "AudioError.hpp"
+
+std::array<char const *, static_cast<size_t>(Sounds::SIZE)> const Audio::soundFilenames
+{{
+  nullptr,
+  "resources/sounds/boyaux1.wav",
+  "resources/sounds/euuuh1.wav"
+}};
+std::array<char const *, static_cast<size_t>(Musics::SIZE)> const Audio::musicFilenames
+{{
+  nullptr,
+  "resources/musics/small_world.ogg"
+}};
+
+Audio Audio::instance;
+
+Audio::Audio()
+{
+  alutInit(nullptr, nullptr);
+  device = alcOpenDevice(nullptr);
+  Audio::checkError();
+}
+
+Audio::~Audio()
+{
+  std::for_each(sounds.cbegin(), sounds.cend(),
+		[](auto &it) { alDeleteBuffers(1, &it.second); });
+  alcCloseDevice(device);
+  alutExit();
+}
+
+Audio &Audio::getInstance(void)
+{
+  return instance;
+}
+
+bool Audio::checkError(bool throws)
+{
+  ALCenum error = alGetError();
+
+  if (error != AL_NO_ERROR)
+    {
+      if (throws)
+	throw AudioError(alutGetErrorString(error));
+      else
+	return false;
+    }
+  return true;
+}
+
+void Audio::clearError(void)
+{
+  alGetError();
+}
+
+ALuint Audio::bufferFromSound(Sounds s)
+{
+  if (s == Sounds::NONE || s == Sounds::SIZE)
```

Comment:
Since you're allready doing bound-checking, you could check all invalid values.
Also FYI, this condition could be optimised.